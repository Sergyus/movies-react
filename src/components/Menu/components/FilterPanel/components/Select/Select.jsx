import React, { useCallback, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from '../../FilterPanel.scss';

const Select = ({
  setActive, setActiveItem, fetchedGenres, fetchMoviesByGenre, filter,
}) => {
  const [selectValue, setSelectValue] = useState('');
  const genres = fetchedGenres;
  const genreList = genres.map(item => (
    <option value={item.id} key={item.id}>{item.name}</option>
  ));
  useEffect(() => {
    if (filter !== 'Genres') {
      setSelectValue('');
    }
  }, [filter, setSelectValue]);
  return (
    <li className={styles.navItem}>
      <select
        onChange={useCallback((e) => {
          setSelectValue(e.target.value);
          setActive('Genres');
          fetchMoviesByGenre(1, e.target.value);
        }, [setActive, fetchMoviesByGenre, setSelectValue])}
        aria-label="Genres"
        value={selectValue}
        className={cn(styles.selectGenre, setActiveItem('Genres'))}
        name="selectGenre"
      >
        <option value="" disabled hidden>Genre</option>
        { genreList }
      </select>
    </li>
  );
};

Select.propTypes = {
  setActive: PropTypes.func,
  setActiveItem: PropTypes.func,
  fetchedGenres: PropTypes.arrayOf(PropTypes.object),
  fetchMoviesByGenre: PropTypes.func,
  filter: PropTypes.string,
};

Select.defaultProps = {
  setActive: () => {},
  setActiveItem: () => {},
  fetchedGenres: [],
  fetchMoviesByGenre: () => {},
  filter: 'Trending',
};

export default Select;
