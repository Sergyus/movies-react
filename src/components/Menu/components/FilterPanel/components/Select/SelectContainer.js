import { connect } from 'react-redux';
import Select from './Select';
import { fetchMoviesByGenre, fetchedGenresSelector, filterSelector } from '../../../../../../modules/app';

const mapStateToProps = state => ({
  fetchedGenres: fetchedGenresSelector(state),
  filter: filterSelector(state),
});

export default connect(mapStateToProps, { fetchMoviesByGenre })(Select);
