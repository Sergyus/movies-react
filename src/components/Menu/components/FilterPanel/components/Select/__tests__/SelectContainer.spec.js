import React from 'react';
import { create } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Select from '..';

describe('Select component', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      genres: [],
    },
  };
  const store = mockStore(initialState);
  it('Select renders correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Select />
      </Provider>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
