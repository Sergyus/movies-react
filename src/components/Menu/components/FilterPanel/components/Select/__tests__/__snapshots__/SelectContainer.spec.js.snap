// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Select renders correctly 1`] = `
<li
  className="navItem"
>
  <select
    aria-label="Genres"
    className="selectGenre"
    name="selectGenre"
    onChange={[Function]}
    value=""
  >
    <option
      disabled={true}
      hidden={true}
      value=""
    >
      Genre
    </option>
  </select>
</li>
`;
